lmfit/parameter.py

```diff
@@ -611,10 +595,10 @@ def _getval(self):
                     self.value = self._expr_eval(self._expr_ast)
                     check_ast_errors(self._expr_eval)
 
-        try:
-            self.value = max(self.min, min(self._val, self.max))
-        except(TypeError, ValueError):
-            self.value = nan
+        v = self._val
+        if v > self.max: v = self.max
```

Comment:
I'd change these lines to:

    if self.min < self._val < self.max:
        pass
    else:
        self._val = max(self.min, min(self._val, self.max))
    self.value = self._val

I like the range check form instead of two ifs. It's also faster.